client/landing/gutenboarding/index.tsx

```diff
@@ -124,9 +124,10 @@ function waitForSelectedSite(): Promise< Site | undefined > {
 		unsubscribe = subscribe( () => {
 			const resolvedSelectedSite = select( SITE_STORE ).getSite( selectedSite );
 			if ( resolvedSelectedSite ) {
-				resolve( resolvedSelectedSite );
+				return resolve( resolvedSelectedSite );
```

Comment:
Explanation: Exit this function so it doesn't get resolved to `undefined` again below.